### PR TITLE
Merge to master

### PR DESCRIPTION
**Pull request by janani-somaskandan**

### CODE REVIEW

File: adaptors/OutputLogger.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package adaptors
```
Comment (aravind-murthy-001) [inline adaptors/OutputLogger.go]: file names are usually in snake case in golang.

File: config/ConfigurationV2.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    Seed_events []string
+}
+
+type EventProbablity struct{
+    Correlation_matrix CorrelationMatrix
+    Independent_events map[string]float64
+}
+
+type UserSegmentV2 struct {
+    Number_of_users int
```
Comment (aravind-murthy-001) [inline config/ConfigurationV2.go]: variables / methods are in camel case.
It is in camel caps if you want to expose it outside the package.

File: operations/dataGenV2.go [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+	var newUserWg sync.WaitGroup
+
+	/*Calculating USERNAME indexing across segments
+	Ex: Segment1 has 10 users and Segment2 has 5 users
+	Segment1 will have users named U1,U2...U10 and 
+	Segment2 will have U11... U15
+	New seeded users will have name from U16*/
+	var userCounter int = 1
+	userIndex := make(map[string]int)
+	for item, element := range config.ConfigV2.User_segments {
```
Comment (aravind-murthy-001) [inline operations/dataGenV2.go]: If I understand right, we are generating all users in one tick. And we would have to define multiple segments for every tick.

better to make this probabilistic too, where we configure saying we will start around 100 users every minute and run it in a loop.

File: operations/dataGenV2.go [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    
+	event := rand.Intn(probMap.EventCorrelationMultiplier[*lastKnownGoodState])
+	eventName, state := probMap.EventCorrelationRangeMap[*lastKnownGoodState].Get(event)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("EventProbablityRangeMapWithCorrelation: Key not found %v", event))
+	}
+	*lastKnownGoodState = eventName
+	return eventName
+}
+
```
Comment (aravind-murthy-001) [inline operations/dataGenV2.go]: A doc comment for each method explaining what the method does will be useful.
https://golang.org/doc/effective_go.html#commentary

File: operations/dataGenV2.go [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+func ComputeRangeMap(probMap map[string]float64) (utils.RangeMap, int) {
```
Comment (aravind-murthy-001) [inline operations/dataGenV2.go]: Further we should add unit tests for all important methods.

File: operations/dataGenV2.go [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+		
+	//TODO call this from util once you find a way to iterate values
+	for _, element := range probMap {
+		if element < min && element != 0{
+			min = element
+		}
+	}
+
+	multiplier := 1.0
+	temp := 0.0
```
Comment (aravind-murthy-001) [inline operations/dataGenV2.go]: multiplier = ceil(1 / min) ?
Also need to handle the case if min is 0.

File: operations/dataGenV2.go [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+	Log.Debug.Printf("Main: %s Completed for user Range %v - %v", segmentName, userRangeStart ,userRangeEnd)
+}
+
+func PreComputeRangeMap(segment config.UserSegmentV2) (SegmentProbMap) {
+
+	var probMap SegmentProbMap
+	probMap.EventCorrelationRangeMap = make(map[string]utils.RangeMap)
+	probMap.EventCorrelationMultiplier = make(map[string]int)
+	for item, element := range segment.Event_probablity_map.Correlation_matrix.Events {
+		probMap.EventCorrelationRangeMap[item], probMap.EventCorrelationMultiplier[item] = ComputeRangeMap(element)
```
Comment (aravind-murthy-001) [inline operations/dataGenV2.go]: Better to manage these two variables in a struct and map the item to the struct.

File: operations/dataGenV2.go (hunk 6 above)
Comment (janani-somaskandan) [reply operations/dataGenV2.go]: Initialization is done with min = 1.0

File: operations/dataGenV2.go (hunk 5 above)
Comment (janani-somaskandan) [reply operations/dataGenV2.go]: Done

File: operations/dataGenV2.go (hunk 4 above)
Comment (janani-somaskandan) [reply operations/dataGenV2.go]: Have already added basic comments. Will see what to be added further

File: operations/dataGenV2.go (hunk 3 above)
Comment (janani-somaskandan) [reply operations/dataGenV2.go]: currently its 1 new user per tick apart from predefined users. will configure it to n

**Pull request merged**